File: python/src/factorial.py
def factorial(num: int):
	assert isinstance(num, int), f"{num} must be integer"

	if num <= 1:
		return num

	return num * factorial(num - 1)


if __name__ == "__main__":
	print(
		factorial(5) # 120
	)

	print(
		factorial(6) # 720
	)

	print(
		factorial(7) # 5040
	)

File: typescript/src/factorial.ts
function getFactorial(num: number): number {
	if (num <= 1) return num

	return num * getFactorial(num - 1)
}

console.log(
	getFactorial(5) // 120
)

console.log(
	getFactorial(6) // 720
)

console.log(
	getFactorial(7) // 5040
)